### PR TITLE
141617721 refactor prework dk

### PR DESCRIPTION
**Pull request by DavidKnott**
Extracts food filter into it's own file and class
Extracts methods for creating, deleting, and validating foods into their own functions
Changes parent().parent() to parentsUntil()

### CODE REVIEW

File: lib/filter_foods.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+$('document').ready(function() {
+  var filter = new FilterFoods();
+  filter.by_name();
+})
+
+class FilterFoods {
+  by_name(){
```
Comment (wlffann) [inline lib/filter_foods.js:7]: Ooooh a class!

File: lib/foods.js [hunk 2]
```diff
@@ -67,14 +41,54 @@ $('document').ready(function() {
     $('#calories-field .validation-error').empty();
   }
 
+function valid_food(name, calories){
+  check_for_name(name)
+  check_for_calories(calories)
+  return (name == "" || calories == "");
+}
+
```
Comment (wlffann) [inline lib/foods.js:49]: Very clean!

**Review by wlffann — COMMENTED** [review]
Thanks for making these refactors! Looks great to me.

File: lib/filter_foods.js (hunk 1 above)
Comment (jdconrad89) [reply lib/filter_foods.js:7]: love the organization of this!

**Comment by jdconrad89** [comment]
@DavidKnott these refactors look Awesome!! Thanks for knocking this out!

**Pull request merged**